test/src/scheduler.js

```diff
@@ -319,4 +337,41 @@ describe('Scheduler', function() {
           assert.equal(tgThree.afterSecondReloadPageLoadCalled, true);
         });
   });
+
+  describe('saves assets when --save-assets is set', function() {
+    let prepareAssetsCalled = false;
+    let saveAssetsCalled = false;
+    const _prepareAssets = assets.prepareAssets;
+    assets.prepareAssets = function() {
+      prepareAssetsCalled = true;
+      return _prepareAssets.apply(this, arguments);
+    };
+    const _saveAssets = assets.saveAssets;
+    assets.saveAssets = function() {
+      saveAssetsCalled = true;
+      return _saveAssets.apply(this, arguments);
+    };
+
+    const screenshotGatherer = new TestScreenshotGatherer();
+    const gatherers = [screenshotGatherer];
+    const options = {
+      driver: fakeDriver,
+      url: 'https://example.com',
+      date: new Date(1464737670547),
+      flags: {
+        saveAssets: true
+      }
+    };
+
+    it('collects artifacts and goes through the scheduler', () => {
+      return scheduler.run(gatherers, options).then(_ => {
+        assert.equal(screenshotGatherer.postProfilingCalled, true);
+      });
+    });
+
+    it('asset saving functions are called from scheduler', () => {
+      assert.ok(prepareAssetsCalled);
+      assert.ok(saveAssetsCalled);
```

Comment:
rather than checking the specific functions were called, can't you just check that the assets were created?